acceptance/tests/options/config_file/ttls/cached_custom_group_core_custom_facts.rb

```diff
@@ -0,0 +1,79 @@
+test_name 'ttls configured with custom group containing core and custom facts' do
+  tag 'risk:high'
+
+  require 'facter/acceptance/user_fact_utils'
+  extend Facter::Acceptance::UserFactUtils
+
+  custom_fact_file = 'custom_facts.rb'
+  custom_fact_name  = 'random_custom_fact'
+  custom_fact_value = 'custom fact value'
+
+  custom_fact_content = <<-CUSTOM_FACT
+  Facter.add(:#{custom_fact_name}) do
+    setcode do
+      Facter.value('system_uptime.seconds')
+    end
+  end
+  CUSTOM_FACT
+
+  config_data = <<~FACTER_CONF
+    facts : {
+      ttls : [
+          { "cached-custom-facts" : 3 days },
+      ]
+    }
+    fact-groups : {
+      cached-custom-facts : ["#{custom_fact_name}", "system_uptime"],
+    }
+  FACTER_CONF
+
+  agents.each do |agent|
+    cache_folder = get_cached_facts_dir(agent['platform'], on(agent, facter('kernelmajversion')).stdout.chomp.to_f)
+    fact_dir = agent.tmpdir('facter')
+    env = { 'FACTERLIB' => fact_dir }
+
+    config_dir = get_default_fact_dir(agent['platform'], on(agent, facter('kernelmajversion')).stdout.chomp.to_f)
+    config_file = File.join(config_dir, 'facter.conf')
+
+    step "Agent #{agent}: create config file" do
+      agent.mkdir_p(config_dir)
+      create_remote_file(agent, config_file, config_data)
+      
+      fact_file = File.join(fact_dir, custom_fact_file)
+      create_remote_file(agent, fact_file, custom_fact_content)
+    
+      teardown do
+        agent.rm_rf(fact_dir)
+        agent.rm_rf("#{cache_folder}/*")
+        agent.rm_rf(config_file)
+      end
+
+      step "should log that it creates cache file and it caches custom facts found in facter.conf" do
+        on(agent, facter("--debug --json", environment: env)) do |facter_result|
+          output_json = JSON.parse(facter_result.stdout.chomp)
+          custom_fact_value = output_json['system_uptime']['seconds']
+          assert_match(/#{custom_fact_value}/, facter_result.stdout.chomp, "#{custom_fact_name} value changed")
```

Comment:
I think that you should do a second request for the cached fact, and compare that value, with what facter returned in the first request